### ISSUE TITLE
Please help, data folder empty using docker compose, how to fix this?

### ISSUE THREAD

**Issue by mkoorn** [jacobalberty/unifi-docker#664]
Hi,

I was looking at my current setup and I see that the data and log folders are empty.

running `docker exec unify ls` shows me al the directories.

My docker-compose contains 
```volumes:
      - '/srv/docker/unify:/unify'
```

I think adding these volumes will fix it.
```
      - /srv/docker/unify/data:/unifi/data
      - /srv/docker/unify/log:/unifi/log
```

But what will happen when I add these to the compose file and then run the container again? Will I lose the data?
Is it possible to first copy just both folders and contents to my host folder and then restarting the container with the adjusted compose.yml?

I also see 2 folders cert and run.
Are these needed for my backup and should I also add these as volumes?

thanks in advance.

**Comment by Fraddles** [comment]
Yes, if you copy those folders to your host volume mount (from inside the container) then they should stay when you bring down the stack.  I am not sure how well that will go with the controller service running...

Another alternative might be to take a full backup from the controller itself, then start with a fresh controller and restore the backup.

In my compose file I have;
```
    volumes:
      - ./data:/unifi
      - ./run:/var/run/unifi
```
The `data` folder has `data`, `log`, and `run` (empty) subfolders.  The additional `run` mount was required to allow the container to run as a non-root user.  Full compose file here; https://github.com/Fraddles/Home-Automation/tree/main/UniFi%20Controller

Cheers.

**Comment by mkoorn** [comment]
Thanks for the input all.

I ended up creating a backup from the controller and also a copy from all the data within the docker.
Brought the docker down, added the volumes and brought it back up again.

Data was gone, but with the controller backup I had the whole config up and running within minutes.
All the config is now available from the volumes specified in the docker compose.
Good stuff!

Thanks again!

**Comment by mkoorn** [comment]
resolved
